pkg/infra/kubernetes/exec_unit.go

```diff
@@ -355,10 +355,11 @@ func (unit *HelmExecUnit) addEnvsVarToDeployment(envVarNames []string) ([]Value,
 
 			deployment.Spec.Template.Spec.Containers[0].Env = append(deployment.Spec.Template.Spec.Containers[0].Env, newEv)
 			values = append(values, Value{
-				ExecUnitName: unit.Name,
-				Kind:         deployment.Kind,
-				Type:         string(EnvironmentVariableTransformation),
-				Key:          v,
+				ExecUnitName:        unit.Name,
+				Kind:                deployment.Kind,
+				Type:                string(EnvironmentVariableTransformation),
+				Key:                 v,
+				EnvironmentVariable: envVar,
```

Comment:
well the reason im including all of it is because our pulumi plugin knows how to process our core.Environment struct as a dependency environment variable and thats the value we need here. The key remainder is for helm to know how to set the values in the install directory, but it doesnt know how to get an rds instances connection string, etc